### PR TITLE
Update README and add script to simplify vendoring openshift/kubernetes

### PR DESCRIPTION
**Pull request by marun**

### CODE REVIEW

**Comment by sjenning** [comment]
should have said LGTM instead of tagging to let others apiserver and controller-manager people to weigh in
/lgtm cancel

File: README.md [hunk 1]
```diff
@@ -5,25 +5,78 @@ Origin Kubernetes
...
+[openshift/kubernetes](https://github.com/openshift/kubernetes)
+fork. Upstream staging repos are used where possible, but some tests
+depends on functionality only present in the fork.
 
-OKD is licensed under the [Apache License, Version 2.0](http://www.apache.org/licenses/).
+### Updating k8s.io/kubernetes vendoring
+
+When bumping the version of `openshift/kubernetes` vendored by this
+repo, it's necessary to update the version of staging repos that are
+also sourced (via a subpath) from `openshift/kubernetes`. A helper
```
Comment (sttts) [inline README.md:43]: this sentence confuses me. How would I "update the version of staging repos" ?

Comment (sttts) [reply README.md:43]: I think it is super important to say which o/k branch we vendor and that all upstream patches must go there first via PR.

Comment (marun) [reply README.md:43]: I say as much beginning on line 20. This section is specific to how to update the vendoring.

Comment (marun) [reply README.md:43]: > all upstream patches must go there first via PR.

I say as much beginning on line 20. This section is specific to how to update the vendoring.

Comment (sttts) [reply README.md:43]: line 20 is about hyperkube, not about origin. Do I miss anything? Which branch do we vendor here in origin master?

Comment (marun) [reply README.md]: The branch names should match between the 2 repos (i.e. master branch of origin should vendor master branch of openshift/kubernetes). I've updated the docs to hopefully provide clarity, PTAL.

File: README.md [hunk 2]
```diff
@@ -5,25 +5,86 @@ Origin Kubernetes
...
+Backports and carries against upstream should be proposed to
+`openshift/kubernetes`. If changes merged to `openshift/kubernetes`
+need to land in `origin`, it will be necessary to follow up with a PR
+to `origin` that bumps the vendoring.
 
-See [HACKING.md](https://github.com/openshift/origin/blob/master/HACKING.md) for more details on developing on Origin including how different tests are setup.
+Branch names are correlated across the 2 repositories such that
+changes merged to a given branch in `openshift/kubernetes` should be
+vendored into the same branch in `origin` (e.g. `master` in
+`openshift/kubernetes` is vendored into `master` in `origin`).
```
Comment (sttts) [inline README.md:30]: I don't think we want that. We should rather use branches like `origin-4.5-kubernetes-1.18.3`. Master branches and release-4.x are only used to build hyperkube. We should not mix hyperkube and origin.

Comment (marun) [reply README.md:30]: As per offline discussion, I've added a note below this section indicating that a transition to vendoring origin-specific branches is pending. PTAL.

**Comment by mfojtik** [comment]
/override ci/prow/images

This is simple documentation fix, does not need CI

**Pull request merged**